lumigator/python/mzai/backend/backend/api/routes/health.py

```diff
@@ -41,24 +42,63 @@ def get_job_metadata(job_id: UUID) -> JobSubmissionResponse:
     except json.JSONDecodeError as e:
         loguru.logger.error(f"JSON decode error: {e}")
         loguru.logger.error(f"Response text: {resp.text}")
+        raise HTTPException(
+            status_code=HTTPStatus.INTERNAL_SERVER_ERROR, detail="Invalid JSON response"
+        ) from e
+
+
+@router.get("/jobs/{job_id}/logs")
+def get_job_logs(job_id: UUID) -> JobLogsResponse:
+    resp = requests.get(urllib.parse.urljoin(settings.ray_jobs, f"{job_id}/logs"))
+
+    if resp.status_code == HTTPStatus.NOT_FOUND:
+        raise HTTPException(
+            status_code=HTTPStatus.NOT_FOUND,
+            detail=f"Job logs for ID: {job_id} not found - {resp.text or ''}",
+        )
+    elif resp.status_code == HTTPStatus.BAD_REQUEST:
+        raise HTTPException(
+            status_code=HTTPStatus.BAD_REQUEST,
+            detail=f"Job ID: {job_id} not submitted via the Ray API - {resp.text or ''}",
+        )
+    elif resp.status_code != HTTPStatus.OK:
+        loguru.logger.error(f"Unexpected status code getting job logs text: {resp.status_code}")
+        loguru.logger.error(f"Upstream job metadata response: {resp}")
         raise HTTPException(
             status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
-            detail="Invalid JSON response"
+            detail=f"Unexpected error getting job logs for ID: {job_id} - {resp.text or ''}",
```

Comment:
Same thing about leaky abstractions.

```suggestion
            detail=f"Unexpected error getting job logs for ID: {job_id}''}",
```

Reply:
Same concern as above

Reply:
Was this missed or was there consensus against changing it?